scripts/in_nic/vocab/pull_data.py

```diff
@@ -0,0 +1,60 @@
+import requests
+import json
+import pandas as pd
+
+url =  "http://vocab.nic.in/rest.php/"
+
+data_files = {
+    'States': 'states/',
+    'Country':  'country/',
+    'Districts': 'district/tn/',
+    'Pincode Details': 'pincode/110003/',
+    'Taluks': 'taluk/tn/',
+    'Language': 'lang/',
+    'Taluks (District - wise)': 'taluk/tn/district/TN003/',
+    'Audiences': 'audience/',
+    'Sectors': 'sectors/',
+    'Sub Sectors': 'sectors/subsectors/',
+    'Site Count': 'sitecount/',
+    'Sub Sectors (Sector-wise)': 'sectors/ST01/subsectors/'
+}
+
+
+class VocabularyDataLoader:
+    def __init__(self, data_files, url):
+        """
+        Reads in the desired url paths to pull from. Initializes
+        dictionary that will store the data frames after pulling.
+
+        :param url: url where the data is stored.
+        :param data_files: dictionary mapping each data type to be
+        extrated to its relative path in the url.
+        """
+        self.paths = {
+            key: url + data_files[key] + 'json' for key in data_files
+        }
+        self.data_frames = {}
+
+    def pull_state_data(self):
+        """
+        Pulls data for states from the url specified at init.
+        """
+        path = self.paths['States']
+        response = json.loads(requests.get(path).text)
+        df = pd.DataFrame(
+        [[state['state'][data_type] for data_type in state['state']] for state in response['states']],
```

Comment:
http://google.github.io/styleguide/pyguide.html#27-comprehensions--generator-expressions

Multiple for clauses or filter expressions are not permitted. Use loops instead when things get more complicated.